src/main/java/Duke.java

```diff
@@ -1,10 +1,303 @@
+     * @param endingIndex Ending index of substring
+     * @return The trimmed substring
+     */
+    private static String getTrimmedSubstring(String input, int startingIndex, int endingIndex) {
+        return input.substring(startingIndex, endingIndex).trim();
+    }
+
+    /**
+     * Print the tasks currently in the list.
+     * @param list The list of tasks
+     * @param listItemCount The number of tasks in the list
+     */
+    private static void listTasks(Task[] list, int listItemCount) {
+        System.out.println("Here are the tasks in your list: ");
+        for (int i = 0; i < listItemCount; i += 1) {
+            System.out.println((i + 1) + NUMBER_LIST_SEPARATOR + list[i]);
+        }
+    }
+
+    // ---------------------------------------------------
+    // PRINTED MESSAGES
+    // ---------------------------------------------------
+
+    /**
+     * Print an error message when input matches none of the commands.
+     */
+    private static void printInvalidCommandMessage() {
+        System.out.println("I'm not sure what you want to do. Check if you've spelled correctly! " +
+                "Type help for a list of commands.");
+    }
+
+    /**
+     * Print a message when there are no tasks in a list.
+     */
+    private static void printNoTasksInListMessage() {
+        System.out.println("There are no tasks in the list!");
+    }
+
+    /**
+     * Prints a message when attempting to mark an already completed task as done.
+     */
+    private static void printTaskAlreadyDoneMessage() {
+        System.out.println("This task has been completed!");
+    }
+
+    /**
+     * Print an error message when attempting to mark a task not on the list as done.
+     */
+    private static void printInvalidTaskNumberMessage() {
+        System.out.println("I can't find that task in the list!");
+    }
+
+    /**
+     * Print an error message when no event date is given.
+     */
+    private static void printNoEventDateErrorMessage() {
+        System.out.println("Please tell me when the event is happening.");
+    }
+
+    /**
+     * Print an error message when no date is given for a deadline.
+     */
+    private static void printNoDueDateErrorMessage() {
+        System.out.println("Please tell me when the task is due.");
+    }
+
+    /**
+     * Print an error message when no task name is given.
+     */
+    private static void printNoTaskErrorMessage() {
+        System.out.println("Please tell me the name of the task.");
+    }
+
+    /**
+     * Print an error message when no event name is given.
+     */
+    private static void printNoEventErrorMessage() {
+        System.out.println("Please tell me the name of the event.");
+    }
+
+    /**
+     * Print a welcome message when starting the program.
+     */
+    private static void printWelcomeMessage() {
+        System.out.println("Hello! I'm Duke, your personal task assistant.");
+        System.out.println("Type help to see what I can do.");
```

Comment:
suggestion:
instead of using two separate println() you can merge both into one by using '\n'.